pkg/apis/serving/v1alpha1/revision_validation.go

```diff
@@ -149,6 +152,37 @@ func (rs *RevisionSpec) Validate(ctx context.Context) *apis.FieldError {
 	return errs
 }
 
+func validateAnnotations(annotations map[string]string) *apis.FieldError {
+	return validatePercentageAnnotationKey(annotations, serving.QueueSideCarResourcePercentageAnnotation)
+}
+
+func validatePercentageAnnotationKey(annotations map[string]string, resourcePercentageAnnotationKey string) *apis.FieldError {
+	if len(annotations) == 0 {
+		return nil
+	}
+
+	v, ok := annotations[resourcePercentageAnnotationKey]
+	if !ok {
+		return nil
+	}
+	value, err := strconv.ParseFloat(v, 32)
+	if err != nil {
+		return &apis.FieldError{
+			Message: fmt.Sprintf("Invalid value %s for annotation %s: %v", v, resourcePercentageAnnotationKey, err),
+			Paths:   []string{resourcePercentageAnnotationKey},
+		}
+	}
+
+	if value <= float64(0) || value > float64(100) {
+		return &apis.FieldError{
```

Comment:
we have a special error reporter for thing to be in the range ( ErrOutOfBoundsValue)